src/main/kotlin/com/ekino/oss/gradle/plugin/quality/QualityPlugin.kt

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2019 ekino (https://www.ekino.com/)
+ */
+
+package com.ekino.oss.gradle.plugin.quality
+
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.plugins.JavaPlugin
+import org.gradle.api.plugins.quality.Checkstyle
+import org.gradle.api.plugins.quality.CheckstyleExtension
+import org.gradle.api.plugins.quality.CheckstylePlugin
+import org.gradle.api.resources.TextResource
+import org.gradle.api.tasks.SourceSetContainer
+import org.gradle.kotlin.dsl.*
+import org.gradle.testing.jacoco.plugins.JacocoPlugin
+import org.gradle.testing.jacoco.plugins.JacocoPluginExtension
+import org.gradle.testing.jacoco.tasks.JacocoReport
+import org.sonarqube.gradle.SonarQubeExtension
+import org.sonarqube.gradle.SonarQubePlugin
+
+class QualityPlugin: Plugin<Project> {
+
+  override fun apply(project: Project) {
+    with(project) {
+
+      // plugins
+      apply<JavaPlugin>()
+      apply<CheckstylePlugin>()
+      apply<JacocoPlugin>()
+      apply<SonarQubePlugin>()
+
+      val sourceSets = the<SourceSetContainer>()
+
+      // configuration
+      configure(listOf(project)) {
+
+        // Checkstyle configuration
+        configure<CheckstyleExtension> {
+          toolVersion = "8.19"
```

Comment:
Hi,
`8.24` please.

Reply:
By the way I'm pretty sure we shouldn't override default version.

Reply:
I have tested : default version of Checkstyle tool for Gradle v5.6.1  is `8.17`.
Do you think i must keep the default value or force to use the `8.24` instead ?

Reply:
We can stick with `8.24`. I've created https://github.com/ekino/gradle-quality-plugin/issues/10 to make it configurable.

Reply:
You're right : we also ship a default 8.24 compatible checkstyle.xml so we have to keep this version. Being able to configure it is great since it's also possible to override the default checkstyle.xml

Reply:
Ok, good. I have set up the `8.24`. It has the consequence of changing a little bit the `checkstyle.xml` because of this issue https://github.com/checkstyle/checkstyle/issues/2116

Reply:
Thanks